File: src/main/java/com/adjorno/billib/rest/db/Chart.java
package com.adjorno.billib.rest.db;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
@Table(name = "CHART")
public class Chart {
    @Id
    @Column(name = "_id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long mId;

    @Column(name = "NAME")
    private String mName;

    @OneToOne
    @JoinColumn(name = "JOURNAL_ID")
    private Journal mJournal;

    @Column(name = "LIST_SIZE")
    private Integer mListSize;

    @Column(name = "START_DATE")
    private String mStartDate;

    public Chart() {
    }

    public Chart(Long id, String name, Journal journal, Integer listSize, String startDate) {
        mId = id;
        mName = name;
        mJournal = journal;
        mListSize = listSize;
        mStartDate = startDate;
    }

    public Long getId() {
        return mId;
    }

    public void setId(Long id) {
        mId = id;
    }

    public String getName() {
        return mName;
    }

    public void setName(String name) {
        mName = name;
    }

    public Journal getJournal() {
        return mJournal;
    }

    public void setJournal(Journal journal) {
        mJournal = journal;
    }

    public Integer getListSize() {
        return mListSize;
    }

    public void setListSize(Integer listSize) {
        mListSize = listSize;
    }

    public String getStartDate() {
        return mStartDate;
    }

    public void setStartDate(String startDate) {
        mStartDate = startDate;
    }

    @Override
    public String toString() {
        return mName;
    }
}

File: src/main/java/com/adjorno/billib/rest/db/Chart.kt
package com.adjorno.billib.rest.db

import com.fasterxml.jackson.annotation.JsonInclude
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table

@Entity
@Table(name = "CHART")
@JsonInclude(JsonInclude.Include.NON_NULL)
data class Chart(
    @Id
    @Column(name = "_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "NAME")
    var name: String? = null,

    @OneToOne
    @JoinColumn(name = "JOURNAL_ID")
    var journal: Journal? = null,

    @Column(name = "LIST_SIZE")
    var listSize: Int? = null,

    @Column(name = "START_DATE")
    var startDate: String? = null,

    @Column(name = "END_DATE")
    var endDate: String? = null
) {
    override fun toString(): String = name.toString()
}
